src/Microsoft.DocAsCode.Plugins/DefaultFileAbstractLayer.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Plugins
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.IO;
+    using System.Linq;
+
+    public class DefaultFileAbstractLayer : IFileAbstractLayer
+    {
+        public bool CanRead => true;
+
+        public bool CanWrite => true;
+
+        public IEnumerable<string> GetAllInputFiles()
+        {
+            var folder = Path.GetFullPath(
+                Environment.ExpandEnvironmentVariables(
+                    EnvironmentContext.BaseDirectory ?? string.Empty));
```

Comment:
yes.

---
In reply to: [92965300](https://github.com/dotnet/docfx/pull/1012#discussion_r92965300) [](ancestors = 92965300)